core/src/test/java/com/linecorp/armeria/client/routing/ZookeeperEndpointGroupTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import static org.junit.Assert.assertEquals;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.ZooDefs;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.linecorp.armeria.client.Endpoint;
+
+import junitextensions.OptionAssert;
+import zookeeperjunit.ZKFactory;
+import zookeeperjunit.ZKInstance;
+import zookeeperjunit.ZooKeeperAssert;
+
+public class ZookeeperEndpointGroupTest implements ZooKeeperAssert, OptionAssert {
+
+    private static final Duration duration = Duration.ofSeconds(5);
+    private final ZKInstance zkInstance = ZKFactory.apply().create();
+    private String znode = "/testEndPoints";
+    ZookeeperEndpointGroup zookeeperEndpointGroup;
+
+    @Override
+    public ZKInstance instance() {
+        return zkInstance;
+    }
+
+    @Before
+    public void start() throws TimeoutException, Throwable {
+        zkInstance.start().result(duration);
+        initialZkConnection();
+
+    }
+
+    @After
+    public void stop() throws TimeoutException, InterruptedException {
+        zkInstance.destroy().ready(duration);
+
+    }
+
+    @Test
+    public void updateEndpoints() {
+        ArrayList<Endpoint> expected = new ArrayList<>();
+        expected.add(Endpoint.of("127.0.0.1", 8001, 2));
+        expected.add(Endpoint.of("127.0.0.1", 8002, 3));
+
+        createOrUpdateZNode(znode, endPointListTobyteArray(expected));
+        assertEquals(zookeeperEndpointGroup.endpoints(), expected);
+
+    }
+
+    public void initialZkConnection() {
+        ArrayList<Endpoint> expected = new ArrayList<>();
```

Comment:
* `ArrayList<Endpoint>` -> `List<EndPoint>`
* And use `ImmutableList.of(Endpoint.of(...), Endpoint.of(...), Endpoint.of(...)`